azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/repository/storage/MessageStorage.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * *
+ *  * Copyright (c) Microsoft Corporation. All rights reserved.
+ *  * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.android.communication.ui.chat.repository.storage
+
+import com.azure.android.communication.ui.chat.models.MessageInfoModel
+import com.azure.android.communication.ui.chat.repository.MessageRepositoryReader
+import com.azure.android.communication.ui.chat.repository.MessageRepositoryWriter
+import com.azure.android.communication.ui.chat.service.sdk.wrapper.ChatMessageType
+import java.util.TreeMap
+import org.threeten.bp.OffsetDateTime
+import org.threeten.bp.ZoneOffset
+
+internal class MessageStorage : MessageRepositoryReader(), MessageRepositoryWriter {
```

Comment:
follow the delegate pattern.

MessageRepositoryTreeReader
MessageRepositoryTreeWriter

add a factory constructor to MessageRepository.

Reply:
Yes. This was an initial draft. Updating before pushing for review.

Reply:
Do we still need this class? It's in the PR, I do see Reader/Writer now.